### ISSUE TITLE
Too many Crashes due to java.lang.OutOfMemoryError

### ISSUE THREAD

**Issue by tomurtogu** [Unity-Technologies/com.unity.mobile.notifications#186]
I see many crashes due to java.lang.OutOfMemoryError on Google Play Vitals
Unity Version: 2020.3.33f
Mobile Notification Package: 2.0.1
Platform: Android
OS : Android 12 (SDK 31)
Devices:
Samsung Galaxy A51
Samsung Galaxy M21
Samsung Galaxy A32 5G
Motorola moto g(30)
Motorola motorola edge 20
Samsung Galaxy A72
Samsung Galaxy S21 5G
Samsung Galaxy M31
Samsung Galaxy Tab S6 Lite
realme C25
Samsung Galaxy A71

Log:

```
java.lang.OutOfMemoryError: 
  at android.os.Parcel.createString8Array (Parcel.java:1489)
  at android.content.pm.ApplicationInfo.<init> (ApplicationInfo.java:1999)
  at android.content.pm.ApplicationInfo.<init> (ApplicationInfo.java:62)
  at android.content.pm.ApplicationInfo$1.lambda$createFromParcel$0 (ApplicationInfo.java:1968)
  at android.content.pm.ApplicationInfo$1$$ExternalSyntheticLambda0.readRawParceled (Unknown Source)
  at android.os.Parcel.readSquashed (Parcel.java:2134)
  at android.content.pm.ApplicationInfo$1.createFromParcel (ApplicationInfo.java:1968)
  at android.content.pm.ApplicationInfo$1.createFromParcel (ApplicationInfo.java:1965)
  at android.os.Parcel.readParcelable (Parcel.java:3345)
  at android.os.Parcel.readValue (Parcel.java:3238)
  at android.os.Parcel.readArrayMapInternal (Parcel.java:3635)
  at android.os.BaseBundle.initializeFromParcelLocked (BaseBundle.java:292)
  at android.os.BaseBundle.unparcel (BaseBundle.java:236)
  at android.os.BaseBundle.getString (BaseBundle.java:1196)
  at com.unity.androidnotifications.UnityNotificationUtilities.deserializeNotificationCustom (UnityNotificationUtilities.java:300)
  at com.unity.androidnotifications.UnityNotificationUtilities.deserializeNotificationIntent (UnityNotificationUtilities.java:219)
  at com.unity.androidnotifications.UnityNotificationUtilities.deserializeNotificationIntent (UnityNotificationUtilities.java:206)
  at com.unity.androidnotifications.UnityNotificationManager.loadNotificationIntents (UnityNotificationManager.java:429)
  at com.unity.androidnotifications.UnityNotificationRestartOnBootReceiver.onReceive (UnityNotificationRestartOnBootReceiver.java:23)
  at android.app.ActivityThread.handleReceiver (ActivityThread.java:4761)
  at android.app.ActivityThread.access$1700 (ActivityThread.java:310)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2292)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8663)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:567)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1135)
```

**Comment by abogarsukov-braingames** [comment]
Just upgraded to v2.0.1. We see a lot of crashes with the same stack trace too. They happen on the very app start. Seems to be Android 12 only.

**Comment by abogarsukov-braingames** [comment]
Just upgraded to v2.0.1. We see a lot of crashes with the same stack trace too. They happen on the very app start. Seems to be Android 12 only.

**Comment by stanislavsmagin** [comment]
Facing the same issue in version 2.0.1. A lot of crashes on Android 12. @aurimasc, do you know when the next version 2.0.2 will be published?

**Comment by abogarsukov-braingames** [comment]
It loos like this has been fixed: https://github.com/Unity-Technologies/com.unity.mobile.notifications/issues/151. The fix missed 2.0.1 and is going to be in 2.0.2.

**Comment by tomurtogu** [comment]
The problem still persists.
